crates/client/src/client/pagination.rs

```diff
@@ -11,7 +11,7 @@ pub struct PaginationRequest<T> {
     /// The cursor returned from a previous query to indicate an offset
     pub cursor: Option<T>,
     /// The number of results to take
-    pub results: usize,
+    pub results: i32,
```

Comment:
Why `i32`? Isn't this a cardinal number, will it ever be negative?

Reply:
It can't be negative, but using `u32` will cause casting from `u32` to `i32` that potentially can truncate the last bit=) So I decided to not support the last bit at all by using `i32`

Reply:
<img width="889" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/76dbfed6-6dd5-4588-b526-711904e82bd5">

We are also trading the readability of the codebase. If we use `u32`, it means that all places from the screenshot will have something like `i32::try_from(results).expect("If the result is more than `i32::MAX` it is impossible to iterate with `PaginationRequest` type")`

Reply:
Right. Ideally you'd want to create an `Unsigned_i32` type that only can be constructed from a valid value. Or better a type that represents the domain case, rather than the implementation, e.g. `ResultCount`, because the caller doesn't care what it's using under the hood, they just want to construct a valid instance.

It all feels like a smell that it's represented as a `i32` at _any_ level though. Do you know why we're using `i32` under the hood and not `u32` or `u16` or something?

Reply:
We do it because of the `async_graphql` API. But it seems under the hood still doesn't accept negative values... 

<img width="826" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/f115c2aa-7487-4cf6-906a-04b6320a803c">

And casts it to `usize` later...=)